### PR TITLE
Configurable Local stack, Tests and CI

### PR DESCRIPTION
**Pull request by martinheidegger**
:wave: My celebratory first PR to a tradle repo, Hi!

### Motivation

This PR has been motivated, and started off by needing a different IP for the `localstack` to enable [a workflow with a remote docker](https://hackmd.io/-GkvvuWeQCKSIslJw9XUXw).

### Main change

In b76b2b92cb858933cab46ae4123c24cae7c76380 I replaced the external `localIp` dependency with a [local code](https://github.com/tradle/aws-utils/commit/b76b2b92cb858933cab46ae4123c24cae7c76380#diff-3705d9084500b5bdbedaf072cac45d4348b1a3b5d3d6eae7063273cd221eb92b) that allows the specification of a `TRADLE_LOCAL_IP` environment variable (in context of the article `10.0.0.1`). In the process I slightly reordered the code from [a regexp that replaces localhost](https://github.com/tradle/aws-utils/compare/master...martinheidegger:configurable-localstack?expand=1#diff-e90d928e01cf78019b8345cd43d08b6096cd8b7707fd4d46af3cf5e3283345baL7) to [template strings](https://github.com/tradle/aws-utils/compare/master...martinheidegger:configurable-localstack?expand=1#diff-090cc26b77705d77fc9b8a67eb8c5d9563220dd1fd640970fdfae3c569924b2cR6) because I generally hat a hard time to figure out where the ip comes from; for future code-readers may have the same issue.

### Extra: Fixing tests

Usually I try provide unit tests but I noticed that the test system had [been disabled at some point in time](https://github.com/tradle/aws-utils/compare/master...martinheidegger:configurable-localstack?expand=1#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L7-R8). So I re-enabled them in fb1750dd019585a649bdb1f4027583f998364d16. Now, the two cli commands `npm run test` and `npm run test:ci` are available both for the repository and a package. The first command `test` will use `ts-node` (run against typescript files), while `test:ci` is ran against the built js files. The tests also use `tape@5` instead of `blue-tape` as async tests have been added to tape and blue-tape is not maintained anymore.

In the process I found the [bucket test](https://github.com/tradle/aws-utils/compare/master...martinheidegger:configurable-localstack?expand=1#diff-c8a17f41a8ad5531dd2bc6f355d5a1a91bef56024803975a52c0c04e246072e8) so I re-enable them to run, though in the process I noticed them broken and [fixed the `destroyBucket` util](00ca130eeb5237158c2e4793797951d9b93093b1) by emptying the bucket before stopping synchronization.

### Extra: Github CI

For the tests to be run automatically on pushes/PR's I also - (with much pain 😓) - added a github action in https://github.com/tradle/aws-utils/commit/2069992c349fd35640860b7dfea14eaba9417158 that should make it easier to determine local development setup issues and automatically test the repo.

See a successful run here: https://github.com/martinheidegger/aws-utils/actions/runs/1103896159

The linking of `lerna` seemed quite odd. As far as I can tell: packages should be linked to parent files so in fb1750dd019585a649bdb1f4027583f998364d16 I also made sure that the lerna packages are linked using `file://...` dependencies. This works smoothly in combination with `lerna bootstrap`. But in the CI process it seems [there is a bug with lerna](https://github.com/lerna/lerna/issues/2352) that I had to workaround on linux by installing the dependencies [for every folder seperately](https://github.com/tradle/aws-utils/compare/master...martinheidegger:configurable-localstack?expand=1#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR31) not sure this is the best setup as I am new to `lerna` but it somehow works...

### Extra: Minor Dev improvements

Going through the code I noticed some smaller things that may help with development.

- Using the local typescript version helps with code hints and linting while editing 78134a200e0b79fd45ce19d7478690242f3ca8e8
- Getting diagnostics on tsc builds helps figure out why they are slow d5681f08dd05fcea34c778cb474066943b9fbabb - for the github action I had to [increase the maximum node memory](https://github.com/tradle/aws-utils/compare/master...martinheidegger:configurable-localstack?expand=1#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR35) to make sure that the code compiles with within the github action. However: since the github action computer tops-out at between 1.5~2Gig any tsc action that exceeds 1.5Gig of memory use will be a slowing down the build time a lot. Currently its at ~560MB but during development when I tried some odd configurations it partially exceeded 1.5gig and then the tests became _really_ slow. Keeping an eye on the build sizes makes sure that the development process may be smooth even on smaller computers. Trying to fix the memory issue I also added `skipLibCheck` as it is supposed to reduce the memory a bit (hard to determine by how much) cce382628e68ad67b934ec3280a07035acdea78c
- Sometimes during testing I accidentally commited `.tsbuildinfo` files that happened through build mistakes. In order to avoid accidental commits, I added them to `.gitignore` 78134a200e0b79fd45ce19d7478690242f3ca8e8

That should be about it. I fear I made a bit an elephant out of a fly but hope this PR works out. 🙇‍♂️

### CODE REVIEW

File: packages/aws-combo/package.json [hunk 1]
```diff
@@ -22,28 +22,22 @@
...
-    "@tradle/aws-s3-client": "^1.0.1-alpha.17",
-    "@tradle/aws-sns-client": "^1.0.1-alpha.17",
+    "@tradle/aws-client-factory": "file:../aws-client-factory",
+    "@tradle/aws-cloudformation-client": "file:../aws-cloudformation-client",
+    "@tradle/aws-cloudwatch-client": "file:../aws-cloudwatch-client",
+    "@tradle/aws-common-utils": "file:../aws-common-utils",
+    "@tradle/aws-iam-client": "file:../aws-iam-client",
+    "@tradle/aws-lambda-client": "file:../aws-lambda-client",
+    "@tradle/aws-s3-client": "file:../aws-s3-client",
+    "@tradle/aws-sns-client": "file:../aws-sns-client",
```
Comment (mvayngrib) [inline packages/aws-combo/package.json:40]: how does resolution work when aws-combo is installed in an external project (e.g. tradle/mycloud)?

Comment (martinheidegger) [reply packages/aws-combo/package.json:40]: It is my understanding that lerna will replace these links on publication.

Comment (mvayngrib) [reply packages/aws-combo/package.json:40]: i didn't know that, cool!

**Review by mvayngrib — APPROVED** [review]
utACK

quick question: what was disableReplication throwing if it ran before emptyBucket?

File: packages/aws-combo/package.json (hunk 1 above)
Comment (martinheidegger) [reply packages/aws-combo/package.json:40]: lerna is a bit finicky about trying out a publish process - the remote needs to be in sync with the repo - but I found this: https://github.com/lerna/lerna/blob/a47fc294393a3e9507a8207a5a2f07648a524722/commands/publish/index.js#L529

**Comment by martinheidegger** [comment]
It was a "BucketNotEmpty" error, which lead me to the reordering. Looking over it in it's cleaned-up state however I am unsure how I caused that error and can not reproduce it. 😔

**Pull request merged**